### PR TITLE
feat(catalog): NearbyPoints bounded radius + distance ordering (CATALOG-3 #943)

### LINKED ISSUE #943: CATALOG-3 — Return bounded nearby Points
## Parent

#936

## Spec card

`CATALOG-3`

## What to build

NearbyPoints owns radius validation, distance ordering, and typed empty results through the PostGIS adapter.

Migrate Catalog and Agent callers; delete duplicate geo SQL/mapping and transport-owned policy.

## Observability

Record radius bucket, count, outcome, and duration without coordinates.

## Acceptance criteria

- [ ] Boundary radii, ordering, empty, and database failure pass; removing the radius bound or distance order fails. [integration]

## Delivery contract

One dedicated worktree and one reviewable PR. Begin with the behavior manifest and characterization commit; land the real local/staging adapter, caller migration, old-path deletion, redacted observability, named mutation proof, and Codecov patch coverage of at least 95%. No compatibility layer.

## Blocked by

- #937 (SAFE-1)

### PR DESCRIPTION
**Pull request by lifeodyssey**
Closes #943 (parent #936). Wave 1.

- `NearbyPoints` use case: radius validation (reject <=0, clamp 50km), deterministic distance ordering (distance asc, id tie-break), typed empty results; single PostGIS adapter (the only geo SQL).
- Deleted duplicate geo SQL/mapping (`src/lib/geo-query.ts`) + transport-owned radius policy.
- 2 named mutations killed (radius bound, distance order). Gates: typecheck, oxlint, 46 files/340 tests, coverage 95.29/82.81/98.46/97.21, spike 31 pass, parity green.

## Summary by Sourcery

Introduce a dedicated NearbyPoints application use case with strict radius policy, deterministic distance ordering, and redacted observability, and wire the catalog nearby API to this use case via new PostGIS and details adapters.

New Features:
- Add NearbyPoints domain use case owning radius validation, distance-based ordering, typed empty results, and observability hooks.
- Introduce outbound adapters for NearbyPoints to handle PostGIS geo queries and point-detail enrichment with a single geo SQL surface.

Bug Fixes:
- Resolve direct-cloud PostGIS geometry parsing issues by replacing interpolated Drizzle fragments with flat parameter binding in the nearby geo adapter.

Enhancements:
- Refactor the nearby API transport to delegate business rules to the NearbyPoints use case and log summarized observations only.
- Align anime-overview and spots APIs to reference the new outbound adapter pattern instead of the removed geo-query helper.

Tests:
- Add unit tests for NearbyPoints covering radius policy, ordering, merging, empty results, error propagation, and observability.
- Add integration spike tests driving NearbyPoints over the real PostGIS adapter and serverless details adapter, and update nearby/spot worker tests to the new wiring.

### CODE REVIEW

File: workers/catalog/src/application/nearby-points.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+/** Deterministic nearest-first order: distance asc, id asc on ties. */
+function sortByDistance(points: NearbyPoint[]): NearbyPoint[] {
+  return [...points].sort(compareDistance);
+}
+
+function compareDistance(left: NearbyPoint, right: NearbyPoint): number {
+  const byDistance = left.distanceM - right.distanceM;
+  if (byDistance !== 0) return byDistance;
+  return left.id < right.id ? -1 : 1;
```
[the remark this answers is not in the document]
Comment (lifeodyssey) [reply workers/catalog/src/application/nearby-points.ts:142]: fixed: compareDistance now returns 0 on equal ids (Array.sort contract) — aa693a25.
